### ISSUE TITLE
error: extent inquiry on scalar

### ISSUE THREAD

**Issue by oroppas** [flang-compiler/f18-llvm-project#844]
`bbc` segfaults on the following code:

```fortran
      program foo

      use comdi

      implicit none
      integer size_head

      size_head = size(head)

      end program foo
```

`comdi.mod` is

```fortran
!mod$ v1 sum:b58c88c8b4ad48b0
module comdi
real(8),allocatable::head(:)
end
```

```
error: loc("./extent_inquiry_on_scalar.f90":8:7): extent inquiry on scalar
LLVM ERROR: aborting
PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace.
Stack dump:
0.	Program arguments: /home/ryuta/packages/flang/build/bin/bbc extent_inquiry_on_scalar.f90
 #0 0x00007f1d83a9fac1 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/home/ryuta/packages/flang/build/bin/../lib/libLLVMSupport.so.13git+0x1e2ac1)
 #1 0x00007f1d83a9d86e llvm::sys::RunSignalHandlers() (/home/ryuta/packages/flang/build/bin/../lib/libLLVMSupport.so.13git+0x1e086e)
 #2 0x00007f1d83aa01f6 SignalHandler(int) Signals.cpp:0:0
 #3 0x00007f1d87aa9870 __restore_rt sigaction.c:0:0
 #4 0x00007f1d834b4d22 raise (/usr/lib/libc.so.6+0x3cd22)
 #5 0x00007f1d8349e862 abort (/usr/lib/libc.so.6+0x26862)
 #6 0x00007f1d839d9662 (/home/ryuta/packages/flang/build/bin/../lib/libLLVMSupport.so.13git+0x11c662)
 #7 0x00007f1d839d94a6 (/home/ryuta/packages/flang/build/bin/../lib/libLLVMSupport.so.13git+0x11c4a6)
 #8 0x00007f1d8778a9d2 (/home/ryuta/packages/flang/build/bin/../lib/libFortranLower.so.13git+0x1db9d2)
 #9 0x00007f1d879c92fe (/home/ryuta/packages/flang/build/bin/../lib/libFortranLower.so.13git+0x41a2fe)
#10 0x00007f1d879c92dd (/home/ryuta/packages/flang/build/bin/../lib/libFortranLower.so.13git+0x41a2dd)
#11 0x00007f1d879c92c6 (/home/ryuta/packages/flang/build/bin/../lib/libFortranLower.so.13git+0x41a2c6)
#12 0x00007f1d879c92b6 (/home/ryuta/packages/flang/build/bin/../lib/libFortranLower.so.13git+0x41a2b6)
#13 0x00007f1d879c9139 (/home/ryuta/packages/flang/build/bin/../lib/libFortranLower.so.13git+0x41a139)
#14 0x00007f1d879c5564 Fortran::lower::readExtent(Fortran::lower::FirOpBuilder&, mlir::Location, fir::ExtendedValue const&, unsigned int) (/home/ryuta/packages/flang/build/bin/../lib/libFortranLower.so.13git+0x416564)
#15 0x00007f1d877fa5e5 (anonymous namespace)::ScalarExprLowering::genval(Fortran::evaluate::DescriptorInquiry const&) ConvertExpr.cpp:0:0
```

**Comment by jeanPerier** [comment]
The issue is that the semantic analysis is building an expression for `size(head)` using the symbol from the module, not the use associated symbol. That is causing issues with lowering that maps the use associated since there should not be references to the directly to the module symbol here (in debug mode, an assert would be hit, it should probably be made a fatal error in release mode too).

In the specific case of the program in the issue, this could be fixed by also mapping the module symbols in lowering, but this is slightly wrong since it can have different volatile and asynchronous attributes and there would be similar issues with symbols in OpenMP regions that could not be solved the same way.

Hence, a patch that fixes how `evaluate::DescriptorInquiry` are built by semantics analysis was pushed for review in llvm: https://reviews.llvm.org/D104385

**Comment by jeanPerier** [comment]
Fixed in fir-dev by #860. Provided program now compiles.

**Comment by oroppas** [comment]
Thanks!
